File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        def dfs(node,depth,arr):
            if not node:
                return
            if depth==len(arr):
                arr.append(node.val)
            else:
                arr[depth]=max(arr[depth], node.val)
            dfs(node.left,depth+1,arr)
            dfs(node.right,depth+1,arr)
        arr=[]
        dfs(root,0,arr)
        return arr

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        List<Integer> arr=new ArrayList<>();
        Queue<TreeNode> queue=new LinkedList<>();
        if(root==null){
            return new ArrayList<>();
        }
        queue.add(root);
        while(!queue.isEmpty()){
            int len=queue.size();
            int max=Integer.MIN_VALUE;
            for(int i=0;i<len;i++){
                TreeNode node=queue.remove();
                max=Math.max(max,node.val);
                if(node.left!=null){
                    queue.add(node.left);
                }
                if(node.right!=null){
                    queue.add(node.right);
                }
            }
            arr.add(max);
        }
        return arr;
    }
}